src/utils/v4_position_planner.rs

```diff
@@ -0,0 +1,141 @@
+            pool.fee,
+            ticks,
+            pool.hooks,
+        )
+        .unwrap_or_default();
+
+        self.planner
+            .add_action(&Actions::MINT_POSITION(MintPositionParams {
+                poolKey: pool_key,
+                tickLower: I24::unchecked_from(tick_lower),
+                tickUpper: I24::unchecked_from(tick_upper),
+                liquidity,
+                amount0Max: amount0_max,
+                amount1Max: amount1_max,
+                owner,
+                hookData: hook_data,
+            }));
+    }
+
+    #[inline]
+    pub fn add_increase(
+        &mut self,
+        token_id: U256,
+        liquidity: U256,
+        amount0_max: u128,
+        amount1_max: u128,
+        hook_data: Bytes,
+    ) {
+        self.planner
+            .add_action(&Actions::INCREASE_LIQUIDITY(IncreaseLiquidityParams {
+                tokenId: token_id,
+                liquidity,
+                amount0Max: amount0_max,
+                amount1Max: amount1_max,
+                hookData: hook_data,
+            }));
+    }
+
+    #[inline]
+    pub fn add_decrease(
+        &mut self,
+        token_id: U256,
+        liquidity: U256,
+        amount0_min: u128,
+        amount1_min: u128,
+        hook_data: Bytes,
+    ) {
+        self.planner
+            .add_action(&Actions::DECREASE_LIQUIDITY(DecreaseLiquidityParams {
+                tokenId: token_id,
+                liquidity,
+                amount0Min: amount0_min,
+                amount1Min: amount1_min,
+                hookData: hook_data,
+            }));
+    }
+
+    #[inline]
+    pub fn add_burn(
+        &mut self,
+        token_id: U256,
+        amount0_min: u128,
+        amount1_min: u128,
+        hook_data: Bytes,
+    ) {
+        self.planner
+            .add_action(&Actions::BURN_POSITION(BurnPositionParams {
+                tokenId: token_id,
+                amount0Min: amount0_min,
+                amount1Min: amount1_min,
+                hookData: hook_data,
+            }));
+    }
+
+    #[inline]
+    pub fn add_settle_pair(&mut self, currency0: &Currency, currency1: &Currency) {
+        self.planner
+            .add_action(&Actions::SETTLE_PAIR(SettlePairParams {
+                currency0: currency0.address(),
+                currency1: currency1.address(),
+            }));
+    }
+
+    #[inline]
+    pub fn add_take_pair(
+        &mut self,
+        currency0: &Currency,
+        currency1: &Currency,
+        recipient: Address,
+    ) {
+        self.planner.add_action(&Actions::TAKE_PAIR(TakePairParams {
+            currency0: currency0.address(),
+            currency1: currency1.address(),
+            recipient,
+        }));
+    }
+
+    #[inline]
+    pub fn add_sweep(&mut self, currency: &Currency, recipient: Address) {
+        self.planner.add_action(&Actions::SWEEP(SweepParams {
+            currency: currency.address(),
+            recipient,
+        }));
+    }
```

Comment:
Use `BaseCurrency` trait and [this](https://github.com/Uniswap/sdks/blob/main/sdks/v4-sdk/src/utils/currencyMap.ts).